docs/pages/database-access/guides/mongodb-self-hosted.mdx

```diff
@@ -0,0 +1,348 @@
+    # Name of the database proxy instance, used to reference in CLI.
+  - name: "example-mongo"
+    # Free-form description of the database proxy instance.
+    description: "Example MongoDB"
+    # Database protocol.
+    protocol: "mongodb"
+    # MongoDB endpoint address or connection string, see below for details.
+    uri: "mongodb://mongo.example.com:27017"
+    # Labels to assign to the database, used in RBAC.
+    static_labels:
+      env: dev
+auth_service:
+  enabled: "no"
+ssh_service:
+  enabled: "no"
+proxy_service:
+  enabled: "no"
+```
+
+<Admonition type="tip" title="Tip">
+  A single Teleport process can run multiple different services, for example
+  multiple database access proxies as well as running other services such an
+  SSH service or an application access proxy.
+</Admonition>
+
+Start the database service:
+
+```sh
+$ teleport start --config=/path/to/teleport-db.yaml --token=/tmp/token
+```
+
+## Standalone vs Replica Set
+
+You can specify either a single connection address as a MongoDB connection URI,
+or a [connection string](https://docs.mongodb.com/manual/reference/connection-string/).
+
+<Tabs>
+  <TabItem label="Standalone">
+  When connecting to a standalone MongoDB server, you can specify the server
+  endpoint:
+
+  ```yaml
+  uri: "mongo.example.com:27017"
+  ```
+
+  Connection string syntax is also accepted:
+
+  ```yaml
+  uri: "mongodb://localhost:27017"
+  ```
+  </TabItem>
+  <TabItem label="Replica set">
+  When connecting to a MongoDB replica set, specify the connection string as
+  a URI:
+
+  ```yaml
+  uri: "mongodb://mongo1.example.com:27017,mongo2.example.com:27017/?replicaSet=rs0"
+  ```
+
+  By default, Teleport will connect to the primary replica set member. If you'd
+  like to connect to a secondary instead, Teleport will respect `readPreference`
+  connection string setting:
+
+  ```yaml
+  uri: "mongodb://mongo1.example.com:27017,mongo2.example.com:27017/?replicaSet=rs0&readPreference=secondary"
+  ```
+
+  Note that if leader election is triggered during the session and the member
+  becomes primary, you won't be auto-reconnected.
+  </TabItem>
+</Tabs>
+
+## Connect
+
+Once the database service has joined the cluster, login to see the available
+databases:
+
+```sh
+$ tsh login --proxy=teleport.example.com:3080 --user=testuser
+$ tsh db ls
+Name          Description     Labels
+------------- --------------- --------
+example-mongo Example MongoDB env=dev
+```
+
+Note that you will only be able to see databases your role has access to. See
+[RBAC](../rbac.mdx) section for more details.
+
+To connect to a particular database server, first retrieve credentials from
+Teleport using `tsh db login` command:
+
+```sh
+$ tsh db login example-mongo
+```
+
+<Admonition type="tip" title="Tip">
+  You can be logged into multiple databases simultaneously.
+</Admonition>
+
+You can optionally specify the database name and the user to use by default
+when connecting to the database instance:
+
+```sh
```

Comment:
I think `bash` and `sh` render the same but not `bsh`. As you know, the values we populate in the code blocks determines the coloring when they render. I've gone through and tried to use `bash` universally just to be sure.

It'll also help us to quickly GREP on all code snippets in the future.